delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/indexing/BlazegraphGlobalEventLog.scala

```diff
@@ -56,7 +55,18 @@ final class BlazegraphGlobalEventLog private (
       .map(_.toMessage)
       .groupWithin(batchMaxSize, batchMaxTimeout)
       .discardDuplicates()
-      .evalMapFilterValue(event => eventExchanges.findFor(event).traverse(_.toState(event, tag).flatMap(_.toGraph)))
+      .evalMapFilterValue { event =>
+        Stream
+          .fromIterator[Task](referenceExchanges.iterator)
+          .evalMap { _(event, tag) }
```

Comment:
shouldn't it be collectFirst? and why are you using a stream at all? Can't you just do a `referenceExchanges.iterator.foldLeftM` instead of all this stream overhead?